week3/src/CardGame.jsx

```diff
@@ -0,0 +1,81 @@
+import img6 from "../public/img/6.png";
+import Header from "./Header";
+import styled from "styled-components";
+import CARD_LIST from "./assets/card";
+import { useState } from "react";
+
+const CardWrapper = styled.div`
+  display: flex;
+  flex-wrap: wrap;
+  padding: 5rem;
+  width: 60rem;
+`;
+const PairCard = styled.div`
+  position: relative;
+  width: 10rem;
+  height: 16rem;
+  padding: 0.5rem;
+`;
+const CardImg = styled.img`
+  object-fit: cover;
+  width: 100%;
+  height: 100%;
+
+  border-radius: 8%;
+`;
+
+const CoverImg = styled(CardImg)`
+  position: absolute;
+  left: 0;
+`;
+//카드 랜덤으로 섞기
+CARD_LIST.sort(() => 0.5 - Math.random());
+
+const CardGame = () => {
+  //카드상태
+  const [isOpen, setOpen] = useState([
+    false,
+    false,
+    false,
+    false,
+    false,
+    false,
+    false,
+    false,
+    false,
+    false,
+  ]);
+  const OpenCard = (index) => {
+    //isOpen[index]를 활용하자..!
+    //console.log(index);
+    //console.log("온클릭 함수 작동");
+    const arr = [...isOpen];
+    //console.log(arr);
+    arr[index] = true;
+    console.log(arr);
+    setOpen(arr);
+
+    //console.log(arr);
+  };
```

Comment:
p3: 이 부분 내가 간단하게 수정해봤어!!
아래처럼 작성하면 조금 더 중복된 코드 작성을 줄이면서 변경할 수 있을 것 같은데, 한 번 생각해보면 좋을 것 같아!

```suggestion  
const [isOpen, setOpen] = useState(Array(10).fill(false));

const OpenCard = (index) => {
  setOpen((prev) => prev.map((item, i) => (i === index ? true : item)));
};
```

Reply:
P4) 도영오빠가 아주 간략하게 잘 줄여줬네요!! 아주 굿굿입니당
코드의 가독성을 위해서 `i` -> `cardIndex`, `index` -> `selectedIndex`로 바꿔주는 것도 고민해보면 좋을 것 같네욤 ㅎㅎ

Reply:
와웅.. 한 줄의 코드였지만,, 엄청나게 공부할 게 많은 코드네요!! 알찬 코드 감사합니다 도영님.. 다은님의 첨언으로 더 잘 이해되었습니다! 모두모두 감사해요감사해요🥰 
아직은 사용할 정도로 이해는 하지 못한 거 같아 반영은 못할 거 같구, 대신 왕초보 스터디 트러블 슈팅에 담아서 추가적으로 공부해보고 추후에 과제 완성하며 수정해보겠습니다.
감사합니당